web/src/main/java/se/inera/intyg/webcert/web/integration/interactions/createdraftcertificate/v3/CreateNewDraftRequestBuilderImpl.java

```diff
@@ -60,13 +58,34 @@ public CreateNewDraftRequest buildCreateNewDraftRequest(Intyg intyg, String inty
         String intygsType = moduleRegistry.getModuleIdFromExternalId(intyg.getTypAvIntyg().getCode());
         Optional<Forifyllnad> forifyllnad = getOptionalForifyllnadIfApplicable(intygsType, intyg.getForifyllnad(), user);
 
-        final Patient patient = TransportConverterUtil.getPatient(intyg.getPatient(), PatientInfo.EXTENDED);
-        patient.setTestIndicator(patientDetailsResolver.isTestIndicator(patient.getPersonId()));
+        final Patient patient = getPatient(intyg);
 
         return new CreateNewDraftRequest(null, intygsType, intygTypeVersion,
             null, hosPerson, patient, intyg.getRef(), forifyllnad);
     }
 
+    private Patient getPatient(Intyg intyg) {
+        final var pnr = intyg.getPatient().getPersonId().getExtension();
+        final var personnummer = Personnummer.createPersonnummer(pnr).orElseThrow();
+        return getPatientFromPU(personnummer);
+    }
+
+    private Patient getPatientFromPU(Personnummer personnummer) {
+        final var personFromPUService = patientDetailsResolver.getPersonFromPUService(personnummer).getPerson();
+        Patient patient = new Patient();
```

Comment:
For consistency you could use final var ;-)

Reply:
Fixed!